kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaConsumerWrapper.java

```diff
@@ -74,7 +75,7 @@
 
   private static final Logger logger = LoggerFactory.getLogger(KafkaConsumerWrapper.class);
 
-  private boolean isAlive = false;
```

Comment:
The stop method cannot guarantee immediate stop either unless you synchronize the stop method with the while block in the separate thread. Here as long as there is no concurrent data write operation, I think it should be fine